include/cutlass/gpu_generics.h

```diff
@@ -314,19 +314,20 @@ using cudaStream_t = void *;
 using dim3 = syclcompat::dim3;
 
 // Atomic
-
-CUTLASS_DEVICE int atomicAdd(int *address, int val) {
+template <typename T>
+CUTLASS_DEVICE T atomicAdd(T *address, T val) {
 #if defined(__SYCL_DEVICE_ONLY__)
-  return syclcompat::atomic_fetch_add(address, val);
+  return syclcompat::atomic_fetch_add<sycl::access::address_space::global_space>(address, val);
```

Comment:
TOL: maybe we can use `is_gmem` or `is_smem` to figure out if the address is a `gmem_ptr` or a `smem_ptr`

Reply:
Checking the pointer type at runtime for global or smem classification generates redundant code in the IGC dumps. I discussed this with @mehdi-goli and we concluded that we can fix this for `gmem` for now.